x-pack/plugins/ml/public/application/services/results_service/results_service.js

```diff
@@ -415,7 +415,7 @@ export function resultsServiceProvider(mlApiServices) {
                   influencerFieldValues: {
                     terms: {
                       field: 'influencer_field_value',
-                      size: maxResults !== undefined ? maxResults : 2,
+                      size: !!maxResults ? maxResults : 2,
```

Comment:
this change is subtle but using a falsey check like `!!` rather than an explicit check for `undefined` changes its behaviour.
if `maxResults` is `0` it would now default to `2`
Not that `0` is a sensible number, it's just generally dangerous IMO using falsey checks for variables that are numbers.

Reply:
I actually had to update this to using `!!` because in the case `maxResults = 0`, the `size` will be 0 and will make the query invalid. 

![image](https://user-images.githubusercontent.com/43350163/98302457-0715a280-1f82-11eb-80ca-5391b20c8c77.png)

Reply:
Updated here https://github.com/elastic/kibana/pull/81923/commits/73555598dd6f29ff6b11152e4a07791cc23fb218